woocommerce/woocommerce-android
# Purchase shipping labels p1

First part of #3698, it implements shipping labels purchase functionality, unit tests will follow in an upcoming PR.

Please note that the commit 671a0f0 is not related to the feature, it's just an improvement to avoid casting from Double to Flow when calling FluxC's methods.

### Screenshots:
<img width=400 src="https://user-images.githubusercontent.com/1657201/113137213-273d4b00-921c-11eb-8d95-b28a07cb9955.png"/> <img width=400 src="https://user-images.githubusercontent.com/1657201/113137226-299fa500-921c-11eb-9a4f-59bf6ae8850c.png"/>
<img width=400 src="https://user-images.githubusercontent.com/1657201/113141404-94071400-9221-11eb-9bb9-a2297da774ec.png"/> <img width=400 src="https://user-images.githubusercontent.com/1657201/113320370-5a5e0800-930a-11eb-85c5-b0be55b8af3b.png" />

### Question
When checking the option "Mark order as complete", we will make another API call to update the order status, to avoid re-purchasing the label if this call fails, I'm just showing an error SnackBar, then continue to the print label screen, WDYT about that @woocommerce/mobile-designers?
<img width=360 src="https://user-images.githubusercontent.com/1657201/113291286-b6189900-92ea-11eb-994a-0a5229dd1ca0.png" />


### Testing:
1. As this needs some API changes that are not released yet, you need to update WCShip using this [file](https://github.com/wordpress-mobile/WordPress-FluxC-Android/files/6210751/woocommerce-services.zip) in your store.
2. Create an order that satisfies shipping label creation.
3. Go to the order detail, and click on Create shipping label.
4. Pass all the steps.
5. Click on Purchase shipping label.
6. Notice that a loading dialog appears.
7. Confirm that the success screen is shown, and that you can print the label.
8. Repeat the steps from 2 to 4.
9. Check the option "Mark order as complete"
10. Click on  Purchase shipping label.
11. Confirm that after going back to order detail, the order's status has been updated to "completed".

PS: I noticed an issue in the label sizes description screen, I'll create a separate issue to check it.

Update release notes:
